tigerpath/majors_and_certificates/majors/MAE_2018.json

```diff
@@ -0,0 +1,419 @@
+{
+  "type": "Major",
+  "name": "Mechanical and Aerospace Engineering",
+  "code": "MAE",
+  "degree": "BSE",
+  "year": 2018,
+  "urls": [
+    "https://ua.princeton.edu/academic-units/department-mechanical-and-aerospace-engineering-0",
+    "https://mae.princeton.edu/undergraduate/programs",
+    "http://mae.princeton.edu/sites/default/files/MAE-UG%20HANDBOOK%202017-2018-Class%202021-%20WEB%20Final.pdf"
+  ],
+  "contacts": [
+    {
+      "type": "Undergraduate Representative",
+      "name": "Michael G. Littman",
+      "email": "mlittman@princeton.edu"
+    },
+    {
+      "type": "Undergraduate Administrator",
+      "name": "Jo Ann Love",
+      "email": "jolove@princeton.edu"
+    },
+    {
+      "type": "Academic Program Assistant",
+      "name": "Theresa Russo",
+      "email": "tar3@princeton.edu"
+    },
+    {
+      "type": "Independent Work Advisor",
+      "name": "Luigi Martinelli",
+      "email": "gigi@phantom2.princeton.edu"
+    }
+  ],
+  "req_list": [
+    {
+      "name": "General Requirements",
+      "max_counted": 1,
+      "min_needed": "ALL",
+      "explanation": "Requirements for study in the department follow the general requirements for the School of Engineering and Applied Science. In addition, the following four courses and one laboratory are normally completed by departmental students before entry into the junior year.\nMechanical and Aerospace Engineering\n206 Introduction to Engineering Dynamics\n221 Thermodynamics\n222 Mechanics of Fluids\n223 Modern Solid Mechanics\n224 Integrated Engineering Science Laboratory\nSome of the above can be satisfied by equivalent courses. For example, students with an interest in structures may take CEE 205 Mechanics of Solids in place of MAE 223; and students with an interest in engineering physics may take PHY 205 Classical Mechanics or PHY 207 Mechanics and Waves in place of MAE 206.\nEach departmental student will be introduced to instrumentation and computer-based data acquisition in the MAE 224 laboratory.\n",
+      "completed_by_semester": 4,
+      "pdfs_allowed": false,
+      "req_list": [
+        {
+          "name": null,
+          "max_counted": null,
+          "min_needed": null,
```

Comment:
Not necessarily. It's enforced by the parent.
Since the parent specifies `"min_needed": "ALL"`, I believe it requires all it's children to as full as possible, which has the same effect.
You can see in the UI that it requires 5 courses (the maximum that can be passed on from the children):
![image](https://user-images.githubusercontent.com/9757500/45338261-252a1880-b55a-11e8-9280-d2ec363b18eb.png)